subprojects/core/src/main/java/org/gradle/internal/service/scopes/ProjectScopeServices.java

```diff
@@ -377,4 +385,12 @@ protected ManagedFactoryRegistry createManagedFactoryRegistry(ManagedFactoryRegi
     protected ModelDefaultsHandler createActionBasedModelDefaultsHandler(SoftwareTypeRegistry softwareTypeRegistry, PluginScheme pluginScheme, InternalProblems problems) {
         return new ActionBasedModelDefaultsHandler(softwareTypeRegistry, pluginScheme.getInspectionScheme(), problems);
     }
+
+    @Provides
+    DefaultProjectLayout createProjectLayout(FileResolver fileResolver, FileCollectionFactory fileCollectionFactory, TaskDependencyFactory taskDependencyFactory,
+                                             FilePropertyFactory filePropertyFactory, Factory<PatternSet> patternSetFactory, PropertyHost propertyHost, FileFactory fileFactory) {
+        File settingsDir = buildLayout.getSettingsDir();
```

Comment:
🤔  Can build layout here be obtained by injection, same as all other parameters?

Reply:
I think it can and that will take care of you other concern too. Testing it.